client/blocks/authentication/social/index.tsx

```diff
@@ -132,7 +132,7 @@ const SocialAuthenticationForm = ( {
 
 						<GithubSocialButton
 							socialServiceResponse={ socialService === 'github' ? socialServiceResponse : null }
-							redirectUri={ getRedirectUri( 'github' ) }
+							redirectUri={ isLogin ? getRedirectUri( 'github' ) : window?.location?.pathname }
```

Comment:
Whether to apply the change at the `GithubSocialButton`'s prop level or as a part of `getRedirectUri()` logic is up for discussion.

Making the proposed change on the prop level (and only on Signup, not Login) is safer - as it will affect just GitHub signups and shouldn't have any effect on logins, nor other SSO options in general.

However, we may decide to make the change in `getRedirectUri()` directly (in `client/blocks/signup-form/social.jsx`) and either apply it to just GitHub SSO or all SSO options.

At this point, I am voting for the safer approach.

We may discuss further once the URI allow-list logic on the backend is updated - when the changes can be tested fully.

Reply:
I will probably make the change inside `getRedirectUri()` at the end.